scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala

```diff
@@ -660,6 +660,7 @@ final class ScannerTokens(val tokens: Tokens)(implicit dialect: Dialect) {
           case RegionTry :: xs
               if !next.isAny[KwCatch, KwFinally] && isLeadingInfix != LeadingInfix.Yes => strip(xs)
           case Nil | (_: CanProduceLF) :: _ => Some(rs)
+          case RegionParen :: RegionForMaybeParens :: _ => Some(rs)
```

Comment:
Looks like this has broken cases like:
```

  test("issue-metals") {
    runTestAssert(
      """|
         |for (m <- decls
         |    if oneCond
         |      && (cond)
         |      && satisfiable) {}
         |
         |""".stripMargin,
      Some("for (m <- decls; if oneCond && cond && satisfiable) {}")
    )(Term.For(
      List(
        Enumerator.Generator(Pat.Var(Term.Name("m")), Term.Name("decls")),
        Enumerator.Guard(Term.ApplyInfix(
          Term.ApplyInfix(
            Term.Name("oneCond"),
            Term.Name("&&"),
            Type.ArgClause(Nil),
            Term.ArgClause(List(Term.Name("cond")), None)
          ),
          Term.Name("&&"),
          Type.ArgClause(Nil),
          Term.ArgClause(List(Term.Name("satisfiable")), None)
        ))
      ),
      Term.Block(Nil)
    ))
  }
```
not sure about the fix here :thinking: